src/OptimalFilter.jl

```diff
@@ -351,22 +351,22 @@ function sample_height_proposal!(height::AbstractArray{T,3},
                                  grid_ext::Grid,
                                  fft_plan::FFTW.FFTWPlan,
                                  fft_plan!::FFTW.FFTWPlan,
-                                 filter_params,
+                                 nprt_per_rank::Int,
                                  rng::Random.AbstractRNG,
                                  obs_noise_std::T,
                                  ) where T
 
-    @assert iseven(filter_params.nprt) "Number of particles must be even"
+    @assert iseven(nprt_per_rank) "Number of particles per rank must be even to use the Optimal Filter"
```

Comment:
:ok_hand: improving  the error message.  Maybe also show the value found?  Something like
```suggestion
    @assert iseven(nprt_per_rank) "Number of particles per rank must be even to use the Optimal Filter, found $(nprt_per_rank)"
```

Reply:
Would be even better if we also showed the rank that found it, but it seems like we are not using MPI here